src/main/flight/alt_hold.c

```diff
@@ -0,0 +1,245 @@
+    return altHoldState->smoothedAltitude;
+}
+
+void altHoldReset(altHoldState_s* altHoldState)
+{
+    simplePidInit(&altHoldState->altPid, -50.0f, 50.0f,
+                  0.01f * altholdConfig()->altPidP,
+                  0.0f,
+                  0.01f * altholdConfig()->altPidI);
+
+    simplePidInit(&altHoldState->velPid, 0.0f, 1.0f,
+                  0.01f * altholdConfig()->velPidP,
+                  0.01f * altholdConfig()->velPidD,
+                  0.0f);
+
+    altHoldState->throttle = 0.0f;
+    altHoldState->enterTime = millis();
+    altHoldState->exitTime = 0;
+    float externalVelocityEstimation = 0.01f * getEstimatedVario();
+    altHoldState->startVelocityEstimationAccel = altHoldState->velocityEstimationAccel - externalVelocityEstimation;
+    altHoldState->targetAltitude = getCurrentAltitude(altHoldState);
+    altHoldState->smoothedAltitude = 0.01f;
+}
+
+void altHoldInit(altHoldState_s* altHoldState)
+{
+    altHoldState->altHoldEnabled = false;
+    altHoldState->throttleFactor = 0.0f;
+    altHoldState->velocityEstimationAccel = 0.0f;
+    altHoldReset(altHoldState);
+}
+
+void altHoldProcessTransitions(altHoldState_s* altHoldState) {
+    bool newAltHoldEnabled = FLIGHT_MODE(ALTHOLD_MODE);
+
+    if (FLIGHT_MODE(GPS_RESCUE_MODE) | failsafeIsActive()) {
+        newAltHoldEnabled = false;
+    }
+
+    if (newAltHoldEnabled && !altHoldState->altHoldEnabled)
+    {
+        altHoldReset(altHoldState);
+    }
+    if (!newAltHoldEnabled && altHoldState->altHoldEnabled) {
+        altHoldState->exitTime = millis();
+    }
+    altHoldState->altHoldEnabled = newAltHoldEnabled;
+
+    uint32_t currTime = millis();
+
+    if (newAltHoldEnabled) {
+        uint32_t timeSinceEnter = currTime - altHoldState->enterTime;
+        if (timeSinceEnter < ALTHOLD_ENTER_PERIOD) {
+            float delta = (float)timeSinceEnter / ALTHOLD_ENTER_PERIOD;
+            altHoldState->throttleFactor = MAX(delta, altHoldState->throttleFactor);
+            return;
+        }
+        altHoldState->throttleFactor = 1.0f;
+        return;
+    }
+
+    if (altHoldState->exitTime == 0) {
+        altHoldState->throttleFactor = 0.0f;
+        return;
+    }
+
+    uint32_t timeSinceExit = currTime - altHoldState->exitTime;
+    if (timeSinceExit < ALTHOLD_MAX_EXIT_PERIOD) {
+        float delta = (float)timeSinceExit / ALTHOLD_MAX_EXIT_PERIOD;
+        altHoldState->throttleFactor = MIN(altHoldState->throttleFactor, 1.0f - delta);
+        return;
+    }
+
+    altHoldState->throttleFactor = 0.0f;
+}
+
+void altHoldUpdate(altHoldState_s* altHoldState)
+{
+    altHoldProcessTransitions(altHoldState);
+
+    float timeInterval = 1.0f / ALTHOLD_TASK_PERIOD;
+
+    float measuredAltitude = getCurrentAltitude(altHoldState);
+
+    t_fp_vector accelerationVector = {{
+        acc.accADC[X],
+        acc.accADC[Y],
+        acc.accADC[Z]
+    }};
+
+    imuTransformVectorBodyToEarth(&accelerationVector);
```

Comment:
Don't we have this value or something similar available in imu.c?

Reply:
I haven't found anything matching.

Reply:
@ctzsnooze : I'm afraid it got removed lately